hook/hook.go

```diff
@@ -10,15 +10,23 @@ import (
 	"github.com/hashicorp/go-multierror"
 )
 
+// Response defines the handler's return value
+// - Sets http.StatusOK and payload if you want the coresponding sink list to process the payload.
+// - Sets other 2xx code if you want to short-circuit the processing, but you don't want to indicate
+//   an error (e.g. skip the processing). You can optionally set a detail to explain the reason.
+//   This will show up on the webhook sender's page.
+// - Sets other HTTP code and err if you do want to indicate an error.
+type Response struct {
+	httpCode int
+	err      error
+	detail   string
```

Comment:
nit: I find it confusing to have both `err` and `detail` at the same time. If we use the status code (as instructed in docstring), then having a dedicated error seems worthless? 

Other than that, I think `message` conveys the meaning better than `detail`.

```suggestion
	message   string
```

Reply:
Good point. Done

Reply:
Done